scripts/check_beaker

```diff
@@ -8,6 +8,33 @@ if [ ! -e "/var/lib/restraint/config.conf" ]; then
         . /etc/profile.d/beaker-harness-env.sh
     fi
 
+    # Try for up to 60 seconds to make sure system has an IP address
+    if command -v hostname; then
+        COUNT=0
+        # 12 tries * 5 seconds = Maximum 60 seconds
+        MAX_TRIES=12
+        # Loop until we have an IP address.
+        # 'hostname -I' will output something like: 192.168.1.1 if we have an
+        # IP address. If no IP address it will output a blank line.
+        IP_FAIL=""
+        while ! hostname -I | grep '[0-9]'
+        do
+            IP_FAIL=0
+            COUNT=$(( COUNT + 1 ))
+            echo "$0: Waiting for IP address. Attempt: ${COUNT} of ${MAX_TRIES} failed" >&2
+            if [[ ${COUNT} -ge ${MAX_TRIES} ]]; then
+                IP_FAIL=1
+                break
+            fi
+            sleep 5.0
+        done
+        if [[ ${IP_FAIL} -eq 0 ]]; then
```

Comment:
Is it possible for the while condition to be try on first try?  If so, ${IP_FAIL} will not be 0.  Why not set IP_FAIL=0 before the loop and remove it from within the loop.

Reply:
> 
> 
> Is it possible for the while condition to be true on first try? If so, ${IP_FAIL} will not be 0. Perhaps set IP_FAIL=0 before the loop and remove it from within the loop.

That is on purpose. If the while is true on the first try then no messages will be output either in the loop or after the loop.

Reply:
John:  I was a little worried about your mixing integer and string with same variable and doing an integer condition check.  I personally don't like integer and string handling in bash since they're handled so differently.  Anyway....I coded similar to what you have above and when IP_FAIL="" the condition if [[ ${IP_FAIL} -eq 0 ]] results in TRUE so you end up echoing IP address acquired.  Perhaps initialize it to -1 if you don't want anything echo'd.  The following is the code I wrote for testing and I simply changed the value of IP_FAIL and ran the script:
#!/usr/bin/bash
IP_FAIL=""
if [[ ${IP_FAIL} -eq 0 ]]; then
    echo "$0: Acquired IP address"
elif [[ ${IP_FAIL} -eq 1 ]]; then
    echo "$0: Failed to acquire IP address"
fi

Reply:
@cbouchar Nice find! 👍 Added to things I've learned about bash 😊

Let me make the suggested change.